quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/PropertyMappingInterceptor.java

```diff
@@ -38,6 +46,57 @@
  */
 public class PropertyMappingInterceptor implements ConfigSourceInterceptor {
 
+    <T> Iterator<T> filterRuntime(Iterator<T> iter, Function<T, String> nameFunc) {
+        if (!isRebuild() && !Environment.isRebuildCheck()) {
+            return iter;
+        }
+        return new Iterator<T>() {
+            T next;
+            boolean isNull;
+
+            @Override
+            public boolean hasNext() {
+                while (next == null && !isNull) {
+                    if (!iter.hasNext()) {
+                        return false;
+                    }
+                    T value = iter.next();
+                    if (!isRuntime(nameFunc.apply(value))) {
+                        next = value;
+                        isNull = next == null;
+                    }
+                }
+                return true;
+            }
+
+            @Override
+            public T next() {
+                if (next == null && !isNull && !hasNext()) {
+                    throw new NoSuchElementException();
+                }
+                T value = next;
+                next = null;
+                isNull = false;
+                return value;
+            }
```

Comment:
Could be simplified as below, right? I don't really like the conditions `next == null && !isNull` as it may seem confusing. Is there any specific case you considered with the functionality?

```suggestion
            T next;

            @Override
            public boolean hasNext() {
                while (next == null) {
                    if (!iter.hasNext()) {
                        return false;
                    }
                    T value = iter.next();
                    if (isRuntime(nameFunc.apply(value))) continue;
                    next = value;
                }
                return true;
            }

            @Override
            public T next() {
                if (next == null && !hasNext()) {
                    throw new NoSuchElementException();
                }
                T value = next;
                next = null;
                return value;
            }
```

Reply:
> There were two reasons for that, the first is that it's not documented whether null is allowed in the iterators - logically it should not be, but it's best not to make assumptions.

+1; once we get null value from the inherited iterator, we should take it into account and not "throw it away". 

> Most of the smallrye logic internally simply creates and accumulates a new collection, rather than filtering over the iterator. However I do see quite a few traversals during build, so it seemed better to filter. But we can certainly just make the logic read like the other interceptors and it will be simpler still.

I like rather filtering it out in the iterator.